File: intel-hex-to-bin.java/src/test/java/demo/FileNameTest.java
package demo;

import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Unit tests for {@link FileName}
 */
public class FileNameTest {

    @Test
    public void getBinFileNameTest() {
        assertEquals(FileName.getBinFileName("test.hex"), "test.bin");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void noNameFailureTest() {
        FileName.getBinFileName(".hex");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void noExtensionFailureTest() {
        FileName.getBinFileName("test");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void emptyExtensionFailureTest() {
        FileName.getBinFileName("test.");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void wrongExtensionFailureTest() {
        FileName.getBinFileName("test.txt");
    }

}

File: intel-hex-to-bin.kotlin/src/test/kotlin/demo/FileNameTest.kt
package demo

import org.testng.Assert.assertEquals
import org.testng.annotations.Test

/**
 * Unit tests for [FileName]
 */
class FileNameTest {

    @Test
    fun getBinFileNameTest() {
        assertEquals(getBinFileName("test.hex"), "test.bin")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun noNameFailureTest() {
        getBinFileName(".hex")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun noExtensionFailureTest() {
        getBinFileName("test")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun emptyExtensionFailureTest() {
        getBinFileName("test.")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun wrongExtensionFailureTest() {
        getBinFileName("test.txt")
    }

}
